app/controllers/cloud_network_controller.rb

```diff
@@ -37,15 +37,20 @@ def button
     when "custom_button"
       custom_buttons
       return
-    when 'instance_compare'
-      return comparemiq
-    when "instance_tag"
-      return tag("VmOrTemplate")
     when "network_router_tag"
       return tag("NetworkRouter")
     when "floating_ip_tag"
       return tag("FloatingIp")
     end
+
+    if params[:pressed].starts_with?('instance_')
```

Comment:
Please, do not test for the prefix. Put a finite list of button IDs here. The next person working with this code will have no idea what are the possible values are.

Aaand the code quality metrics will go down :-D

![](http://3.bp.blogspot.com/-ilMjE1Gh3Yg/VpUAmd-6TWI/AAAAAAAAAbg/-FJ08zxN42s/s1600/WFTPM.png)